### PR TITLE
Reconcile rules for `predicate_id`

### LINKED ISSUE #300: Reconciliation rule: using `predicate_id`
Precedence rules provided confidence is equal. 
- owl:equivalentClass
 - owl:equivalentProperty
 - rdfs:subClassOf 
 - rdfs:subPropertyOf
 - owl:sameAs
 - skos:exactMatch
 - skos:closeMatch
 - skos:broadMatch
 - skos:narrowMatch
 - oboInOwl:hasDbXref
 - skos:relatedMatch
 - rdfs:seeAlso

### LINKED ISSUE #304: parse embedded mode: flag true and default true makes no sense
```
"-E",
    "--embedded-mode",
    default=True,
    is_flag=True,
```

How could it theoretically be switched off?

### PR DESCRIPTION
**Pull request by hrshdhgd**
Fixes #300 

 - [x] Reconcile MSDF with an additional rule: Provided the `confidence` values are equal, `predicate_id` follows the following hierarchy:

```
    # owl:equivalentClass
    # owl:equivalentProperty
    # rdfs:subClassOf
    # rdfs:subPropertyOf
    # owl:sameAs
    # skos:exactMatch
    # skos:closeMatch
    # skos:broadMatch
    # skos:narrowMatch
    # oboInOwl:hasDbXref
    # skos:relatedMatch
    # rdfs:seeAlso
```
- [x] Fix #304

### CODE REVIEW

**Comment by matentzn** [comment]
@hrshdhgd sorry about that my mind isnt where it should be this week. can we review this together on Monday?

**Comment by hrshdhgd** [comment]
Not a problem at all! No hurries.

**Review by matentzn — CHANGES_REQUESTED** [review]

File: tests/test_reconcile.py [hunk 1]
```diff
@@ -13,25 +13,25 @@ class TestReconcile(unittest.TestCase):
 
     def setUp(self) -> None:
         """Test up the test case with the third basic example."""
-        self.msdf = parse_sssom_table(data_dir / "basic3.tsv")
+        self.msdf = parse_sssom_table(data_dir / "basic7.tsv")
```
[the remark this answers is not in the document]
Comment (hrshdhgd) [reply tests/test_reconcile.py]: Done!

File: tests/test_reconcile.py [hunk 2]
```diff
@@ -13,25 +13,35 @@ class TestReconcile(unittest.TestCase):
...
     def test_merge(self):
         """Test merging two tables."""
-        msdf1 = parse_sssom_table(data_dir / "basic.tsv")
-        msdf2 = parse_sssom_table(data_dir / "basic2.tsv")
-        merged_msdf = merge_msdf(msdf1, msdf2)
+        msdf3 = parse_sssom_table(data_dir / "basic.tsv")
+        merged_msdf1 = merge_msdf(self.msdf1, msdf3)
 
-        self.assertEqual(122, len(merged_msdf.df))
+        self.assertEqual(98, len(merged_msdf1.df))
```
Comment (matentzn) [inline tests/test_reconcile.py:38]: Is this expected?

File: tests/test_merge.py [hunk 3]
```diff
@@ -21,8 +21,8 @@ def setUp(self) -> None:
     def test_merge_multiple_inputs(self):
         """Test merging of multiple msdfs."""
         merged_msdf = merge_msdf(*self.msdfs)
-        self.assertEqual(123, len(merged_msdf.df))
+        self.assertEqual(170, len(merged_msdf.df))
```
Comment (matentzn) [inline tests/test_merge.py:24]: very very big change. Is this expected?

Comment (matentzn) [reply tests/test_merge.py:24]: merge_msdf should not reconcile by default.. Is it?

File: tests/test_merge.py [hunk 4]
```diff
@@ -21,8 +21,8 @@ def setUp(self) -> None:
...
 
     def test_merge_single_input(self):
         """Test merging when a single msdf is provided."""
-        self.assertEqual(93, len(merge_msdf(self.msdf).df))
+        self.assertEqual(91, len(merge_msdf(self.msdf).df))
```
Comment (matentzn) [inline tests/test_merge.py:28]: Expected?

File: tests/test_merge.py (hunk 3 above)
Comment (hrshdhgd) [reply tests/test_merge.py:24]: reconcile is True by default.

File: tests/test_reconcile.py (hunk 2 above)
Comment (hrshdhgd) [reply tests/test_reconcile.py:38]: reconcile effect

File: tests/test_merge.py (hunk 4 above)
Comment (hrshdhgd) [reply tests/test_merge.py:28]: reconcile

File: tests/test_merge.py (hunk 3 above)
Comment (matentzn) [reply tests/test_merge.py:24]: Ok it should not be, can we make it false? during a simple merge, no information should be lost. Make sure that all the API methods are also not set to TRUE by default..

Comment (hrshdhgd) [reply tests/test_merge.py]: Done, test numbers will change for `test_reconcile` and `test_merge`.

**Review by matentzn — APPROVED** [review]
Great thanks!

**Pull request merged**